### ISSUE TITLE
Issue with installing on 64bit Windows, Python3, Anaconda

### ISSUE THREAD

**Issue by jo-le** [originell/jpype#198]
Hi, I have downloaded the package source code and through command prompt tried running:

`python setup.py install`

However, I get errors. This is my build printout:

  ```
running install
  running bdist_egg
  running egg_info
  writing top-level names to JPype1.egg-info\top_level.txt
  writing dependency_links to JPype1.egg-info\dependency_links.txt
  writing requirements to JPype1.egg-info\requires.txt
  writing JPype1.egg-info\PKG-INFO
  reading manifest file 'JPype1.egg-info\SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no previously-included files matching '*.pyc' found anywhere in distrib
  ution
  no previously-included directories found matching 'test\classes'
  writing manifest file 'JPype1.egg-info\SOURCES.txt'
  installing library code to build\bdist.win-amd64\egg
  running install_lib
  running build_py
  running build_ext
  setup.py:164: FeatureNotice: Turned ON Numpy support for fast Java array access
    FeatureNotice)
  building '_jpype' extension
  C:\Anaconda3\Scripts\gcc.bat -mdll -O -Wall -DWIN32=1 -DHAVE_NUMPY=1 -Inative\co
  mmon\include -Inative\python\include -Inative\jni_include -IC:\Anaconda3\lib\sit
  e-packages\numpy\core\include -IC:\Anaconda3\include -IC:\Anaconda3\include -c n
  ative\common\jp_array.cpp -o build\temp.win-amd64-3.4\Release\native\common\jp_a
  rray.o
  In file included from native\common\include/jp_utility.h:20:0,
                   from native\common\include/jpype.h:92,
                   from native\common\jp_array.cpp:17:
  native\jni_include/jni.h:74:0: warning: "JNICALL" redefined [enabled by default]

  In file included from native\common\jp_array.cpp:17:0:
  native\common\include/jpype.h:52:0: note: this is the location of the previous d
  efinition
  C:\Anaconda3\Scripts\gcc.bat -mdll -O -Wall -DWIN32=1 -DHAVE_NUMPY=1 -Inative\co
  mmon\include -Inative\python\include -Inative\jni_include -IC:\Anaconda3\lib\sit
  e-packages\numpy\core\include -IC:\Anaconda3\include -IC:\Anaconda3\include -c n
  ative\common\jp_arrayclass.cpp -o build\temp.win-amd64-3.4\Release\native\common
  \jp_arrayclass.o
  In file included from native\common\include/jp_utility.h:20:0,
                   from native\common\include/jpype.h:92,
                   from native\common\jp_arrayclass.cpp:17:
  native\jni_include/jni.h:74:0: warning: "JNICALL" redefined [enabled by default]

  In file included from native\common\jp_arrayclass.cpp:17:0:
  native\common\include/jpype.h:52:0: note: this is the location of the previous d
  efinition
  C:\Anaconda3\Scripts\gcc.bat -mdll -O -Wall -DWIN32=1 -DHAVE_NUMPY=1 -Inative\co
  mmon\include -Inative\python\include -Inative\jni_include -IC:\Anaconda3\lib\sit
  e-packages\numpy\core\include -IC:\Anaconda3\include -IC:\Anaconda3\include -c n
  ative\common\jp_class.cpp -o build\temp.win-amd64-3.4\Release\native\common\jp_c
  lass.o
  In file included from native\common\include/jp_utility.h:20:0,
                   from native\common\include/jpype.h:92,
                   from native\common\jp_class.cpp:17:
  native\jni_include/jni.h:74:0: warning: "JNICALL" redefined [enabled by default]

  In file included from native\common\jp_class.cpp:17:0:
  native\common\include/jpype.h:52:0: note: this is the location of the previous d
  efinition
  C:\Anaconda3\Scripts\gcc.bat -mdll -O -Wall -DWIN32=1 -DHAVE_NUMPY=1 -Inative\co
  mmon\include -Inative\python\include -Inative\jni_include -IC:\Anaconda3\lib\sit
  e-packages\numpy\core\include -IC:\Anaconda3\include -IC:\Anaconda3\include -c n
  ative\common\jp_classbase.cpp -o build\temp.win-amd64-3.4\Release\native\common\
  jp_classbase.o
  In file included from native\common\include/jp_utility.h:20:0,
                   from native\common\include/jpype.h:92,
                   from native\common\jp_classbase.cpp:17:
  native\jni_include/jni.h:74:0: warning: "JNICALL" redefined [enabled by default]

  In file included from native\common\jp_classbase.cpp:17:0:
  native\common\include/jpype.h:52:0: note: this is the location of the previous d
  efinition
  C:\Anaconda3\Scripts\gcc.bat -mdll -O -Wall -DWIN32=1 -DHAVE_NUMPY=1 -Inative\co
  mmon\include -Inative\python\include -Inative\jni_include -IC:\Anaconda3\lib\sit
  e-packages\numpy\core\include -IC:\Anaconda3\include -IC:\Anaconda3\include -c n
  ative\common\jp_env.cpp -o build\temp.win-amd64-3.4\Release\native\common\jp_env
  .o
  In file included from native\common\include/jp_utility.h:20:0,
                   from native\common\include/jpype.h:92,
                   from native\common\jp_env.cpp:17:
  native\jni_include/jni.h:74:0: warning: "JNICALL" redefined [enabled by default]

  In file included from native\common\jp_env.cpp:17:0:
  native\common\include/jpype.h:52:0: note: this is the location of the previous d
  efinition
  C:\Anaconda3\Scripts\gcc.bat -mdll -O -Wall -DWIN32=1 -DHAVE_NUMPY=1 -Inative\co
  mmon\include -Inative\python\include -Inative\jni_include -IC:\Anaconda3\lib\sit
  e-packages\numpy\core\include -IC:\Anaconda3\include -IC:\Anaconda3\include -c n
  ative\common\jp_field.cpp -o build\temp.win-amd64-3.4\Release\native\common\jp_f
  ield.o
  In file included from native\common\include/jp_utility.h:20:0,
                   from native\common\include/jpype.h:92,
                   from native\common\jp_field.cpp:17:
  native\jni_include/jni.h:74:0: warning: "JNICALL" redefined [enabled by default]

  In file included from native\common\jp_field.cpp:17:0:
  native\common\include/jpype.h:52:0: note: this is the location of the previous d
  efinition
  C:\Anaconda3\Scripts\gcc.bat -mdll -O -Wall -DWIN32=1 -DHAVE_NUMPY=1 -Inative\co
  mmon\include -Inative\python\include -Inative\jni_include -IC:\Anaconda3\lib\sit
  e-packages\numpy\core\include -IC:\Anaconda3\include -IC:\Anaconda3\include -c n
  ative\common\jp_invocationhandler.cpp -o build\temp.win-amd64-3.4\Release\native
  \common\jp_invocationhandler.o
  In file included from native\common\include/jp_utility.h:20:0,
                   from native\common\include/jpype.h:92,
                   from native\common\jp_invocationhandler.cpp:1:
  native\jni_include/jni.h:74:0: warning: "JNICALL" redefined [enabled by default]

  In file included from native\common\jp_invocationhandler.cpp:1:0:
  native\common\include/jpype.h:52:0: note: this is the location of the previous d
  efinition
  C:\Anaconda3\Scripts\gcc.bat -mdll -O -Wall -DWIN32=1 -DHAVE_NUMPY=1 -Inative\co
  mmon\include -Inative\python\include -Inative\jni_include -IC:\Anaconda3\lib\sit
  e-packages\numpy\core\include -IC:\Anaconda3\include -IC:\Anaconda3\include -c n
  ative\common\jp_javaenv.cpp -o build\temp.win-amd64-3.4\Release\native\common\jp
  _javaenv.o
  In file included from native\common\include/jp_utility.h:20:0,
                   from native\common\include/jpype.h:92,
                   from native\common\jp_javaenv.cpp:17:
  native\jni_include/jni.h:74:0: warning: "JNICALL" redefined [enabled by default]

  In file included from native\common\jp_javaenv.cpp:17:0:
  native\common\include/jpype.h:52:0: note: this is the location of the previous d
  efinition
  native\common\jp_javaenv.cpp: In member function 'void JPJavaEnv::DeleteLocalRef
  (jobject)':
  native\common\jp_javaenv.cpp:143:2: error: cast from 'jobject {aka _jobject*}' t
  o 'long int' loses precision [-fpermissive]
  native\common\jp_javaenv.cpp: In member function 'void JPJavaEnv::DeleteGlobalRe
  f(jobject)':
  native\common\jp_javaenv.cpp:155:2: error: cast from 'jobject {aka _jobject*}' t
  o 'long int' loses precision [-fpermissive]
  native\common\jp_javaenv.cpp: In member function '_jobject* JPJavaEnv::NewLocalR
  ef(jobject)':
  native\common\jp_javaenv.cpp:167:2: error: cast from 'jobject {aka _jobject*}' t
  o 'long int' loses precision [-fpermissive]
  native\common\jp_javaenv.cpp:171:2: error: cast from 'jobject {aka _jobject*}' t
  o 'long int' loses precision [-fpermissive]
  native\common\jp_javaenv.cpp: In member function '_jobject* JPJavaEnv::NewGlobal
  Ref(jobject)':
  native\common\jp_javaenv.cpp:179:2: error: cast from 'jobject {aka _jobject*}' t
  o 'long int' loses precision [-fpermissive]
  native\common\jp_javaenv.cpp:183:2: error: cast from 'jobject {aka _jobject*}' t
  o 'long int' loses precision [-fpermissive]
  native\common\jp_javaenv.cpp: In member function '_jobject* JPJavaEnv::NewDirect
  ByteBuffer(void*, jlong)':
  native\common\jp_javaenv.cpp:287:2: error: cast from 'jobject {aka _jobject*}' t
  o 'long int' loses precision [-fpermissive]
  error: command 'C:\\Anaconda3\\Scripts\\gcc.bat' failed with exit status 1
```

Do you have any idea why this would be failing?!

My `conda info` is:

```
  Current conda install:

                 platform : win-64
            conda version : 4.2.12
        conda is private : False
        conda-env version : 4.2.12
      conda-build version : 1.8.2
           python version : 3.4.5.final.0
         requests version : 2.11.1
         root environment : C:\Anaconda3  (writable)
      default environment : C:\Anaconda3
         envs directories : C:\Anaconda3\envs
            package cache : C:\Anaconda3\pkgs
```

**Comment by jo-le** [comment]
Sorry, didn't see I should be using

`conda install -c conda-forge jypep1`

**Comment by marscher** [comment]
no problem. If the error occurs with the git master, it is definitely something we should look into.

**Comment by jo-le** [comment]
Hey,

When I do: `conda install -c conda-forge jypep1`

it is forcing me to also install `conda-env-2.6.0` package, which is crashing and for some reason i can't install it. Is JPype not available `conda-env-4.2.13`?

**Comment by marscher** [comment]
this is another issue. conda-env is no direct dependency of jpype, so you should be able to downgrade it again.
